awswrangler/s3/__init__.py

```diff
@@ -20,6 +20,7 @@
 __all__ = [
     "copy_objects",
     "merge_datasets",
+    "merge_upsert_table",
```

Comment:
Could be a minor artifact from a merge - did we not remove it?

Reply:
Currently in `main`, this method is in the [docs](https://aws-sdk-pandas.readthedocs.io/en/stable/stubs/awswrangler.s3.merge_upsert_table.html#awswrangler.s3.merge_upsert_table) but not in the [__init__](https://github.com/aws/aws-sdk-pandas/blob/main/awswrangler/s3/__init__.py) so I was just fixing that.

On a different note, I am keen to remove or deprecate this method but perhaps it should be a separate PR